### PR TITLE
Update code for creating new user by API

### PR DESCRIPTION
**Pull request by jinety**

### CODE REVIEW

File: practice-three/db.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+  "userDatabase": [
```
Comment (phuocduong-agilityio) [inline practice-three/db.json]: For follow RESTful convention [here](https://restfulapi.net/resource-naming/)

File: practice-three/src/js/script.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ const emailInput = document.getElementById('email');
 const salaryInput = document.getElementById('salary');
 const cityInput = document.getElementById('city');
 const submitBtn = document.getElementById('submitBtn');
+const userTableBody = document.getElementById('userTableBody')
+let courseApi = 'http://localhost:3000/userDatabase'
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js]: Should be const

Comment (hieungo-agilityio) [inline practice-three/src/js/script.js]: Why did you use `let`? And this api is related to User, `courseApi` doesn't make sense.

File: practice-three/db.json (hunk 1 above)
Comment (hieungo-agilityio) [inline practice-three/db.json]: `users` is enough.

File: practice-three/src/js/script.js [hunk 3]
```diff
@@ -187,31 +173,45 @@ const renderUserTable = () => {
...
+  const salary = salaryInput.value;
+  const city = cityInput.value;
+  const formData = {
+    name: name,
+    email: email,
+    salary: salary,
+    city: city
+  };
+
+  createUsers(formData);
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js]: Create single user or multiple users?

File: practice-three/src/js/script.js [hunk 4]
```diff
@@ -187,31 +173,45 @@ const renderUserTable = () => {
...
+ * Function POST data to userDatabase 
  */
+ const createUsers = (data, callback) => {
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(data)
+  };
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js:188]: After const, emtpy line

File: practice-three/src/js/script.js [hunk 5]
```diff
@@ -187,31 +173,45 @@ const renderUserTable = () => {
...
+  fetch(courseApi, options)
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js]: courseApi???

File: practice-three/src/js/script.js [hunk 6]
```diff
@@ -187,31 +173,45 @@ const renderUserTable = () => {
...
-      }
-    });
-  });
 }
 
 /**
- * Submit form
+ * Function POST data to userDatabase 
  */
+ const createUsers = (data, callback) => {
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js]: What does callback do?

File: practice-three/src/js/script.js [hunk 7]
```diff
@@ -187,31 +173,45 @@ const renderUserTable = () => {
...
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(data)
+  };
+  fetch(courseApi, options)
+    .then(function(response) {
+      response.json();
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js]: What does response.json() do?

File: practice-three/src/js/script.js [hunk 8]
```diff
@@ -187,31 +173,45 @@ const renderUserTable = () => {
...
+    })
+    .then(callback);
+}
+
+const handleCreateForm = () => {
+  const name = fullNameInput.value;
+  const email = emailInput.value;
+  const salary = salaryInput.value;
+  const city = cityInput.value;
+  const formData = {
```
Comment (hieungo-agilityio) [inline practice-three/src/js/script.js]: Please check https://attacomsian.com/blog/javascript-object-property-shorthand

File: practice-three/src/js/script.js [hunk 9]
```diff
@@ -4,6 +4,8 @@ const emailInput = document.getElementById('email');
 const salaryInput = document.getElementById('salary');
 const cityInput = document.getElementById('city');
 const submitBtn = document.getElementById('submitBtn');
+const userTableBody = document.getElementById('userTableBody')
+const userApi = ' http://localhost:3000/users'
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js]: Redundant space before http:...

File: practice-three/db.json [hunk 10]
```diff
@@ -48,6 +48,20 @@
...
+    {
+      "name": "Van Hoa",
+      "email": "vhoa1000@gmail.com",
+      "salary": "90",
+      "city": "Quang Nam",
+      "id": 9
     }
   ]
-}
+}
```
Comment (hieungo-agilityio) [inline practice-three/db.json]: Add empty line at the end of file

File: practice-three/README.md [hunk 11]
```diff
@@ -42,23 +42,19 @@ Before you continue, ensure you meet the following requirements:
...
   ```
   npm start
   ```
-* Step 8: Open browser with the link below 
+* Step 7: Open browser with the link below
   ```
   http://localhost:1234
   ```
-
+
```
Comment (hieungo-agilityio) [inline practice-three/README.md]: Remove space.

File: practice-three/src/js/script.js [hunk 12]
```diff
@@ -178,34 +178,29 @@ const renderUserTable = () => {
...
   const options = {
     method: 'POST',
     headers: {
       'Content-Type': 'application/json'
     },
     body: JSON.stringify(data)
   };
-  fetch(courseApi, options)
+
+  fetch(userApi, options)
```
Comment (hieungo-agilityio) [inline practice-three/src/js/script.js:191]: No need to handle error of this API call?

File: practice-three/src/js/script.js (hunk 7 above)
Comment (jinety) [reply practice-three/src/js/script.js]: parses JSON response into native JavaScript objects

File: practice-three/src/js/script.js (hunk 4 above)
Comment (jinety) [inline practice-three/src/js/script.js:189]: fixed

File: practice-three/src/js/script.js [hunk 13]
```diff
@@ -4,6 +4,8 @@ const emailInput = document.getElementById('email');
 const salaryInput = document.getElementById('salary');
 const cityInput = document.getElementById('city');
 const submitBtn = document.getElementById('submitBtn');
+const userTableBody = document.getElementById('userTableBody')
```
Comment (phuocduong-agilityio) [inline practice-three/src/js/script.js]: Missing ; line 6,7

File: practice-three/src/js/script.js (hunk 12 above)
Comment (jinety) [reply practice-three/src/js/script.js:191]: fixed

**Pull request merged**